src/sagemaker/workflow/function_step.py

```diff
@@ -308,13 +305,9 @@ def _to_json_get(self) -> JsonGet:
 
         return JsonGet(
             s3_uri=Join(
-                "/",
-                [
-                    s3_root_uri,
-                    ExecutionVariables.PIPELINE_NAME,
-                    ExecutionVariables.PIPELINE_EXECUTION_ID,
-                    self._step.name,
-                    RESULTS_FOLDER,
+                on="/",
+                values=[
+                    get_step(self)._properties.OutputDataConfig.S3OutputPath,
```

Comment:
will this be backwards compatible? as this S3OutputPath will not exist on the steps created with older version

Reply:
I have been thinking about this and here's what I think:
- this SDK change only affects "creating" a pipeline definition and the **JsonGet expr would appear in the regular step arguments that reference function steps**.
- it's the backend to parse the JsonGet expression appeared in the regular step argument definition
- as I've kept the existing JsonGet S3Uri logics in backend without any modification, the backend should be able to parse the older version of step definition. This can be proved by the existing backend integ test which passed in the latest alpha integ test.

Does this make sense?